spatialpandas/geometry/multipolygon2d.py

```diff
@@ -0,0 +1,182 @@
+
+    def to_shapely(self):
+        """
+        Convert to shapely shape
+
+        Returns:
+            shapely MultiPolygon shape
+        """
+        import shapely.geometry as sg
+        polygon_arrays = np.asarray(self.data)
+
+        polygons = []
+        for polygon_array in polygon_arrays:
+            ring_arrays = [line_coords.reshape(len(line_coords) // 2, 2)
+                           for line_coords in polygon_array]
+
+            rings = [sg.LinearRing(ring_array) for ring_array in ring_arrays]
+            polygons.append(sg.Polygon(shell=rings[0], holes=rings[1:]))
+
+        return sg.MultiPolygon(polygons=polygons)
+
+    @classmethod
+    def from_shapely(cls, shape, orient=True):
+        """
+        Build a spatialpandas MultiPolygon2d object from a shapely shape
+
+        Args:
+            shape: A shapely Polygon or MultiPolygon shape
+            orient: If True (default), reorder polygon vertices so that outer shells
+                    are stored in counter clockwise order and holes are stored in
+                    clockwise order.  If False, accept vertices as given. Note that
+                    some algorithms will not behave properly if the above ordering
+                    convention is not followed, so only set orient=False if it is
+                    known that this conventions is followed in the input data.
+        Returns:
+            spatialpandas MultiPolygon2d
+        """
+        shape_parts = cls._shapely_to_coordinates(shape, orient)
+        return cls(shape_parts)
+
+
+    @property
+    def boundary(self):
+        new_offsets = self.buffer_offsets[1]
+        new_data = pa.ListArray.from_arrays(new_offsets, self.buffer_values)
+        return MultiLine2d(new_data)
+
+    @property
+    def length(self):
+        return compute_line_length(self._values, self._value_offsets)
+
+    @property
+    def area(self):
+        return compute_area(self._values, self._value_offsets)
+
+
+class MultiPolygon2dArray(GeometryArray):
+    _element_type = MultiPolygon2d
+    _nesting_levels = 3
+
+    @property
+    def _dtype_class(self):
+        return MultiPolygon2dDtype
+
+    @classmethod
+    def from_geopandas(cls, ga, orient=True):
+        """
+        Build a spatialpandas MultiPolygon2dArray from a geopandas GeometryArray or
+        GeoSeries.
+
+        Args:
+            ga: A geopandas GeometryArray or GeoSeries of MultiPolygon or
+                Polygon shapes.
+            orient: If True (default), reorder polygon vertices so that outer shells
+                    are stored in counter clockwise order and holes are stored in
+                    clockwise order.  If False, accept vertices as given. Note that
+                    some algorithms will not behave properly if the above ordering
+                    convention is not followed, so only set orient=False if it is
+                    known that this conventions is followed in the input data.
```

Comment:
```suggestion
                    known that this convention is followed in the input data.
```

Reply:
Maybe explain somewhere that `orient=True` can be costly (as otherwise it's unclear why anyone would ever turn it off)?